File: android/app/src/main/java/org/fossasia/openevent/utils/SharedPreferencesUtil.java
package org.fossasia.openevent.utils;

import android.content.Context;
import android.content.SharedPreferences;

import org.fossasia.openevent.OpenEventApp;

public class SharedPreferencesUtil {
    private static SharedPreferences sharedPreferences;
    private static SharedPreferences.Editor editor;

    static {
        sharedPreferences = OpenEventApp.getAppContext().getSharedPreferences(ConstantStrings.FOSS_PREFS, Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
    }

    //Add more methods if needed

    public static void putInt(String key, int value) {
        editor.putInt(key, value).apply();
    }

    public static int getInt(String key, int defaultValue) {
        return sharedPreferences.getInt(key, defaultValue);
    }

    public static void putLong(String key, long value) {
        editor.putLong(key, value).apply();
    }

    public static long getLong(String key, long defaultValue) {
        return sharedPreferences.getLong(key, defaultValue);
    }

    public static void putString(String key, String value) {
        editor.putString(key, value).apply();
    }

    public static String getString(String key, String defaultValue) {
        return sharedPreferences.getString(key, defaultValue);
    }

    public static void putBoolean(String key, boolean value) {
        editor.putBoolean(key, value).apply();
    }

    public static boolean getBoolean(String key, boolean defaultValue) {
        return sharedPreferences.getBoolean(key, defaultValue);
    }

    public static void remove(String key) {
        editor.remove(key).apply();
    }
}

File: android/app/src/main/java/org/fossasia/openevent/common/utils/SharedPreferencesUtil.kt
package org.fossasia.openevent.common.utils

import android.content.SharedPreferences
import android.preference.PreferenceManager

import org.fossasia.openevent.OpenEventApp

object SharedPreferencesUtil {
    private val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(OpenEventApp.getAppContext())
    private val editor: SharedPreferences.Editor = sharedPreferences.edit()

    //Add more methods if needed
    @JvmStatic
    fun putInt(key: String, value: Int) {
        editor.putInt(key, value).apply()
    }

    @JvmStatic
    fun getInt(key: String, defaultValue: Int): Int {
        return sharedPreferences.getInt(key, defaultValue)
    }

    @JvmStatic
    fun putLong(key: String, value: Long) {
        editor.putLong(key, value).apply()
    }

    @JvmStatic
    fun getLong(key: String, defaultValue: Long): Long {
        return sharedPreferences.getLong(key, defaultValue)
    }

    @JvmStatic
    fun putString(key: String, value: String) {
        editor.putString(key, value).apply()
    }

    @JvmStatic
    fun getString(key: String, defaultValue: String?): String? {
        return sharedPreferences.getString(key, defaultValue)
    }

    @JvmStatic
    fun putBoolean(key: String, value: Boolean) {
        editor.putBoolean(key, value).apply()
    }

    @JvmStatic
    fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return sharedPreferences.getBoolean(key, defaultValue)
    }

    @JvmStatic
    fun remove(key: String?) {
        editor.remove(key).apply()
    }
}
